frappe/__init__.py

```diff
@@ -650,6 +651,22 @@ def clear_last_message():
 		local.message_log = local.message_log[:-1]
 
 
+class Throw(Exception):
```

Comment:
Can we instead handle this inside `throw` itself. It's confusing to have `frappe.throw` and `frappe.Throw`.

`throw` already has `exc` argument.

Reply:
Also things that look like function calls shouldn't be `PascalCase`.

Reply:
To cut out some unnecessary traceback and have better representation of error causes, I figured the `raise ... from...` would have to be close to the source of the exception. I ssem.not to have found a way to reliably manipulate the traceback otherwise.

It _may_ be possible, I wasn't just able to, left alone reliably so.

So ultimately iteration along those lines led me to believe that a class inheriting from `Exception` is the (only?) way.